tools/jbrowse2/jbrowse2.py

```diff
@@ -0,0 +1,1619 @@
+    try:
+        if len(node.findall("dataset")) != 1:
+            # exit early
+            return metadata
+    except Exception:
+        return {}
+
+    for (key, value) in node.findall("dataset")[0].attrib.items():
+        metadata["dataset_%s" % key] = value
+
+    if node.findall("history"):
+        for (key, value) in node.findall("history")[0].attrib.items():
+            metadata["history_%s" % key] = value
+
+    if node.findall("metadata"):
+        for (key, value) in node.findall("metadata")[0].attrib.items():
+            metadata["metadata_%s" % key] = value
+        # Additional Mappings applied:
+        metadata[
+            "dataset_edam_format"
+        ] = '<a target="_blank" href="http://edamontology.org/{0}">{1}</a>'.format(
+            metadata["dataset_edam_format"], metadata["dataset_file_ext"]
+        )
+        metadata["history_user_email"] = '<a href="mailto:{0}">{0}</a>'.format(
+            metadata["history_user_email"]
+        )
+        metadata["hist_name"] = metadata["history_display_name"]
+        metadata[
+            "history_display_name"
+        ] = '<a target="_blank" href="{galaxy}/history/view/{encoded_hist_id}">{hist_name}</a>'.format(
+            galaxy=GALAXY_INFRASTRUCTURE_URL,
+            encoded_hist_id=metadata.get("history_id", "not available"),
+            hist_name=metadata.get("history_display_name", "not available"),
+        )
+    if node.findall("tool"):
+        for (key, value) in node.findall("tool")[0].attrib.items():
+            metadata["tool_%s" % key] = value
+        metadata[
+            "tool_tool"
+        ] = '<a target="_blank" href="{galaxy}/datasets/{encoded_id}/show_params">{tool_id}{tool_version}</a>'.format(
+            galaxy=GALAXY_INFRASTRUCTURE_URL,
+            encoded_id=metadata.get("dataset_id", ""),
+            tool_id=metadata.get("tool_tool_id", ""),
+            tool_version=metadata.get("tool_tool_version", ""),
+        )
+    return metadata
+
+
+class JbrowseConnector(object):
+    def __init__(self, outdir, jbrowse2path):
+        self.assemblies = []  # these require more than a few line diff.
+        self.assmeta = {}
+        self.giURL = GALAXY_INFRASTRUCTURE_URL
+        self.outdir = outdir
+        self.genome_firstcontig = None
+        self.jbrowse2path = jbrowse2path
+        os.makedirs(self.outdir, exist_ok=True)
+        self.genome_names = []
+        self.trackIdlist = []
+        self.tracksToAdd = {}
+        self.config_json = {}
+        self.config_json_file = os.path.join(outdir, "config.json")
+        self.clone_jbrowse()
+
+    def get_cwd(self, cwd):
+        if cwd:
+            return self.outdir
+        else:
+            return subprocess.check_output(["pwd"]).decode("utf-8").strip()
+            # return None
+
+    def subprocess_check_call(self, command, output=None, cwd=True):
+        if output:
+            log.debug("cd %s && %s >  %s", self.get_cwd(cwd), " ".join(command), output)
```

Comment:
We should probably not expose the CWD here. So we need to make sure the tool never runs in debug mode on production instances.

Reply:
No problem for me - that was from Helena's ancient code so yes please - remove.